src/datasets/arrow_dataset.py

```diff
@@ -5225,6 +5225,7 @@ def push_to_hub(
         self,
         repo_id: str,
         config_name: str = "default",
+        as_default_config_name: Optional[bool] = None,
```

Comment:
Maybe name it `set_default_config` (boolean) ?

Reply:
OK for the rename.

On the other hand, for the type, I decided to make it default None, because I find it weird to set the default config_name "default" but set the default set_default_config False...

Reply:
Right, makes sense !

Reply:
What about `is_default_config`? That way we make clear is boolean...

Reply:
I like `set_xxx` because it's explicit that it would unset the previous default config

Reply:
Normally `set_xxx` would be used to set a value for the variable `xxx`.

Anyway, I set it as you said.

Reply:
Well technically we do set `default: True` ^^'

Reply:
But maybe it should be named `set_default` then instead of `set_default_config`

Reply:
Done.